### PR TITLE
Add more documentation, now that we have published packages

### LINKED ISSUE #6: README for Repo
This repo doesn't have any docs yet. We should add some.

## Expected Behavior

- Instructions how to:
  - Generate types, what is outputted.
  - How to use `build.sh` and `generate.sh`
    - And what these programs do (what files are outputted).
- Prerequisites:
  - OS, other library versions required (got an error `Unknown OSTYPE: darwin19`)

## Actual Behavior

Not much detail.

---

Some starting guidance here would help with the other repos.

### PR DESCRIPTION
**Pull request by jskeet**
Fixes #6 as far as we need to for now.

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+do this; the proto-based classes in [src/Google.Events.Protobuf]() will be
+regenerated.
+
+If event data types or properties have been added, the tests for
+other serialization frameworks will fail; the tests should currently
+be fixed by manually adding the corresponding types and properties.
+
+Note: this is currently only supported on Linux and Windows; if you
+are developing on another platform, please edit `generate-protos.sh`
+appropriately, and create a pull request wit the changes. (It's just
```
Comment (amanda-tarafa) [inline BUILDING.md:23]: nit: wit => with

Comment (jskeet) [reply BUILDING.md:23]: Whoops, thanks.

**Comment by grant** [comment]
@jskeet This PR has 4 links that 404. 1 is a NuGet link that might be in-flight, but we should fix the other links.

**Comment by jskeet** [comment]
@grant: Thanks, will fix.

**Pull request merged**